packages/rum/src/domain/record/startFullSnapshots.ts

```diff
@@ -65,17 +67,31 @@ export function startFullSnapshots(
     return records
   }
 
-  fullSnapshotCallback(takeFullSnapshot())
+  fullSnapshotReadyCallback(takeFullSnapshot())
 
+  let cancelIdleCallback: (() => void) | undefined
   const { unsubscribe } = lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, (view) => {
     flushMutations()
-    fullSnapshotCallback(
-      takeFullSnapshot(view.startClocks.timeStamp, {
-        shadowRootsController,
-        status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
-        elementsScrollPositions,
-      })
-    )
+    function takeSubsequentFullSnapshot() {
+      flushMutations()
+      fullSnapshotReadyCallback(
+        takeFullSnapshot(view.startClocks.timeStamp, {
+          shadowRootsController,
+          status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
+          elementsScrollPositions,
+        })
+      )
+    }
+
+    if (isExperimentalFeatureEnabled(ExperimentalFeature.ASYNC_FULL_SNAPSHOT)) {
+      if (cancelIdleCallback) {
+        cancelIdleCallback()
+      }
+      fullSnapshotPendingCallback()
+      cancelIdleCallback = requestIdleCallback(takeSubsequentFullSnapshot)
+    } else {
+      takeSubsequentFullSnapshot()
+    }
   })
 
   return {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Make sure to call `cancelIdleCallback` in the `stop()` function
```ts
return {
  stop: () => {
    unsubscribe()
    cancelIdleCallback?.()
  }
}
```